chat-core/src/OpenAiEmbedFunc.ts

```diff
@@ -4,16 +4,9 @@ import { EmbedFunc } from "./EmbedFunc";
 import { logger } from "./services/logger";
 import { stripIndent } from "common-tags";
 import { CreateEmbeddingResponse } from "openai";
+import { backOff, BackoffOptions } from "exponential-backoff";
 
-/**
-  Creates an OpenAI implementation of the embedding function.
- */
-export const makeOpenAiEmbedFunc = ({
-  baseUrl,
-  deployment,
-  apiKey,
-  apiVersion,
-}: {
+export type MakeOpenAiEmbedFuncArgs = {
```

Comment:
i think there's be dependencies in the server that will break b/c of this change. did you rerun the Q&A server tests with this change?

Reply:
Yes and it still works because the new param is optional